src/parsers/manifest/dash/indexes/base.ts

```diff
@@ -145,18 +147,21 @@ export default class BaseRepresentationIndex implements IRepresentationIndex {
    * @param {Object} index
    * @param {Object} context
    */
-  constructor(index : IBaseIndexIndexArgument, context : IBaseIndexContextArgument) {
+  constructor(index : IBaseIndexIndexArgument,
+              context : IBaseIndexContextArgument,
+              isContentFragmented: boolean) {
     const { periodStart,
             periodEnd,
             representationBaseURL,
             representationId,
             representationBitrate } = context;
     const { timescale } = index;
-
+    this._isContentFragmented = isContentFragmented;
     const presentationTimeOffset = index.presentationTimeOffset != null ?
       index.presentationTimeOffset : 0;
 
-    const indexTimeOffset = presentationTimeOffset - periodStart * timescale;
+    const realTimescale = (timescale != null ? timescale : 1);
```

Comment:
Who provide the timescale information? Why is it not defined in some cases?
It might be better to ensure at the earliest possible time that such values are defined

Reply:
Still